packages/main/src/plugin/updater.spec.ts

```diff
@@ -0,0 +1,165 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { beforeEach, expect, test, vi } from 'vitest';
+import type { MessageBox } from '/@/plugin/message-box.js';
+import type { StatusBarRegistry } from '/@/plugin/statusbar/statusbar-registry.js';
+import type { CommandRegistry } from '/@/plugin/command-registry.js';
+import { Disposable } from '/@/plugin/types/disposable.js';
+import { UPDATER_UPDATE_AVAILABLE_ICON } from '/@/plugin/index.js';
+import { Updater } from '/@/plugin/updater.js';
+
+const mocks = vi.hoisted(() => ({
+  // MessageBoxMock
+  showMessageBoxMock: vi.fn(),
+  // StatusBarRegistry
+  setEntryMock: vi.fn(),
+  // CommandRegistry
+  registerCommandMock: vi.fn(),
+  executeCommandMock: vi.fn(),
+  // Electron App
+  getVersionMock: vi.fn(),
+  // Electron Update
+  downloadUpdateMock: vi.fn(),
+  quitAndInstallMock: vi.fn(),
+  checkForUpdatesMock: vi.fn(),
+  onMock: vi.fn(),
+  // utils
+  isLinuxMock: vi.fn(),
+}));
```

Comment:
hello, I'm not sure to understand why we need this block ?

if you do
```ts
vi.mock('electron-updater', () => ({
  autoUpdater: {
    downloadUpdate: vi.fn()
})
```

Reply:
and then `vi.mocked(...)` to access the mocked method

Reply:
I have a preference for this method, of declaring all the mocks in a single places, and accessing them through `mocks.[name]`

We see often some `vi.spy`, of `vi.mock` but when we want to set default implementation in the beforeEach combined with a `vi.resetAllMocks();` it is easier to access those mock isntance through a single place, rather than calling several methods.

There are in the code, also some tests declaring `MockInstance` objects, in multiple places, in AI-Studio we were in favour of using this single mocks objects to keep them in a single places

Reply:
they'll still be declared at a single place

currently you're losing type by doing that